ls1intum/Artemis
# `Lectures`: Allow sorting of lectures in the course management

### Checklist
#### General


#### Client

### Motivation and Context


Resolves [#6439](https://github.com/ls1intum/Artemis/issues/6439)
This feature would make it easier for users to find specific lectures, especially in courses with a large number of lectures. The sorting feature would enhance the user experience by providing a more organized view of lectures.

### Description

A lecture sorting feature is added in the course management section of Artemis. This feature allows users to sort lectures based on different criteria such as id, title, description, visible from, start, and end date. 
- Alphabetical sorting for string values adapted into existing sort service
- Course column removed

### Steps for Testing

Prerequisites:
- 1 Instructor/Admin/Editor/TA (who has access to course management)
- Course which has multiple lectures

1. Log in to Artemis
2. Navigate to Course Management
3. Navigate Lectures
4. Verify sorting lectures by id, title, description, visible from, start, and end date possible.


### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [lecture.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5115/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture.component.ts.html) | 96.47% | ✅ |



### Screenshots


![sort_lecture_v3](https://github.com/ls1intum/Artemis/assets/11020241/ac4f02f3-c5e0-4820-8f8b-9364216ecec9)
